packages/pillarbox-debug-panel/src/graph-component.js

```diff
@@ -0,0 +1,116 @@
+   *                                              Older data points are removed as new ones are added.
+   * @param {string} [options.fillStyle='rgba(11,83,148)'] - The color of the bars on the graph.
+   * @param {string} [options.strokeStyle='rgb(50,50,50)'] - The color of the stroke around the line on the graph.
+   */
+  constructor(player, options = {}) {
+    super(player, options);
+    this.data = [];
+  }
+
+  /**
+   * Constructs the DOM element that will be used to render the graph.
+   * It ensures the element is a canvas and sets up the necessary properties.
+   *
+   * @param {string} [tag='canvas'] - The HTML tag name for the element. Must be 'canvas'.
+   * @param {Object} [props={}] - Additional properties to set on the element.
+   * @param {Object} [attributes={}] - Additional attributes to set on the element.
+   * @returns {HTMLCanvasElement} The created canvas element.
+   */
+  createEl(tag = 'canvas', props = {}, attributes = {}) {
+    if (tag !== 'canvas') {
+      log.error(`Creating a GraphComponent with an HTML element of ${tag} is not supported; the element must be a 'canvas'`);
+    }
+
+    props = Object.assign({ className: this.buildCSSClass(), }, props);
+
+    return videojs.dom.createEl(tag, props, attributes);
+  }
+
+  /**
+   * Updates the graph with a new data point.
+   *
+   * This method adds a new data point to the graph. If the maximum number of
+   * data points is exceeded, the oldest point is removed. After updating the
+   * data, the graph is redrawn.
+   *
+   * @param {number} value - The new data point to add to the graph.
+   */
+  update(value) {
+    if (this.data.length >= this.options_.maxDataPoints) {
+      this.data.shift();
+    }
+
+    this.data.push(value);
+    this.drawGraph();
+  }
+
+  /**
+   * Draws the graph on the canvas element.
+   *
+   * This method iterates over the stored data points and renders them on the
+   * canvas as a series of bars. Each bar represents a data point, and the
+   * height of the bar is scaled relative to the maximum value in the dataset.
+   */
+  drawGraph() {
+    const ctx = this.el_.getContext('2d');
+    const maxValue = Math.max(...this.data);
+
+    ctx.clearRect(0, 0, this.el_.width, this.el_.height);
+
+    const barWidth = this.el_.width / this.options_.maxDataPoints;
+
+    this.data.forEach((point, index) => {
+      const x = index * barWidth;
+      const y = this.el_.height - (point / maxValue) * this.el_.height; // Scale dynamically based on maxValue
+      const barHeight = (point / maxValue) * this.el_.height;
+
+      ctx.beginPath();
+      ctx.rect(x, y, barWidth, barHeight);
+      ctx.closePath();
+
+      ctx.fillStyle = this.options_.fillStyle;
+      ctx.fill();
+
+      ctx.strokeStyle = this.options_.strokeStyle;
+      ctx.stroke();
+    });
+  }
+
+  buildCSSClass() {
+    return `${super.buildCSSClass()} graph-component`;
+  }
+}
+
```

Comment:
If we add a `typedef` we can use it as the constructor param, what do you think?